### PR TITLE
Add type annotations and fix some lint warnings in Python code

### PR DESCRIPTION
**Pull request by cyyever**
### Description

More robust type checking in Python.
### Motivation and Context
For better code

### CODE REVIEW

**Comment by cyyever** [comment]
@justinchuby More fixes.

File: onnx/reference/ops/_helpers.py [hunk 1]
```diff
@@ -32,16 +32,15 @@ def build_registered_operators_any_domain(
         if isinstance(class_type, type(build_registered_operators_any_domain)):
             continue
         try:
-            issub = issubclass(class_type, OpRun)
+            if issubclass(class_type, OpRun):
```
Comment (xadupre) [inline onnx/reference/ops/_helpers.py:35]: I usually avoid putting in a try catch more than the code meant to fail. The lines you moved are not meant to fail and if they fail, it is a bug we should fix.

File: onnx/reference/ops/aionnx_preview_training/_op_list.py [hunk 2]
```diff
@@ -4,28 +4,24 @@
...
 from onnx.reference.ops._helpers import build_registered_operators_any_domain
-from onnx.reference.ops.aionnx_preview_training._op_run_training import OpRunTraining
 from onnx.reference.ops.aionnx_preview_training.op_adagrad import Adagrad
 from onnx.reference.ops.aionnx_preview_training.op_adam import Adam
 from onnx.reference.ops.aionnx_preview_training.op_momentum import Momentum
 
 
-def _build_registered_operators() -> dict[str, dict[int | None, OpRunTraining]]:
-    return build_registered_operators_any_domain(globals().copy())  # type: ignore[return-value]
+def _build_registered_operators() -> dict[str, dict[int | None, type[OpRun]]]:
```
Comment (xadupre) [inline onnx/reference/ops/aionnx_preview_training/_op_list.py:16]: Why not use OpRunTraining?

Comment (cyyever) [reply onnx/reference/ops/aionnx_preview_training/_op_list.py:16]: Because type checking doesn't allow overriding parent type with sub type.

**Review by justinchuby — APPROVED** [review]

**Comment by justinchuby** [comment]
cc @xadupre for approval

**Review by xadupre — APPROVED** [review]

**Pull request merged**